### DISCUSSION TITLE
Source IP and subnet for Synology

### DISCUSSION THREAD

**Discussion by henryhm0** [teslamate-org/teslamate#1876] [Q&A]
Hi guys, I want to turn Synology firewall to stop all WAN incoming requests.. which IP addresses/ports do I need to open for Tesla mate to work?

**Comment by sit** [comment]
TeslaMate does not require any inbound ports to collect data.

If you're running TeslaMate via Docker on your Synology, you can view TeslaMate by visiting http://your-synology-ip:4000/ when inside your network.

You _could_ expose port 3000 and 4000 from your firewall to the Synology host, but that is not secure.

If you wish to access TeslaMate from outside your private network, you will need to adapt the instructions from the advanced guides such as https://docs.teslamate.org/docs/guides/traefik or perhaps #1758 to provide secure remote access.

**Reply by henryhm0** [reply]
I want to access it internally only.. when I block all external IPs it stops working. I allowed 172.18.0.0 and all ports but that didn’t help. 

![68F78B50-1449-4531-9106-924A5F85D0D9](https://user-images.githubusercontent.com/85140974/127758397-7d6141c1-4b4d-48f8-9ec8-b6da7d877c26.jpeg)

**Comment by henryhm0** [comment]
Looked at which IP docker uses; it uses 172.18.0.1 as a gateway with subnet 172.18.0.0/16. I added an exclusion rule to allow subnet with IP address 172.0.0.0 and subnet mask/prefix length: 255.255.0.0 and this seems to fix my problem.. thanks again for your help :)

**Comment by henryhm0** [comment]
oops i can now access tesla mate on port 4000 but can not access Grafana.. rrgh :)
